CMakeLists.txt

```diff
@@ -93,9 +93,17 @@ macro(PUBCOMP comp)
     add_dependencies(publish ${comp}-pub)
 endmacro(PUBCOMP)
 
+
 # use prefix length to trim path for logging, see inc_internal/utils.h __FILENAME__
-string(LENGTH "${CMAKE_SOURCE_DIR}/" SOURCE_PATH_SIZE)
-add_definitions("-DSOURCE_PATH_SIZE=${SOURCE_PATH_SIZE}")
+if (CMAKE_GENERATOR STREQUAL "Ninja" AND CMAKE_BINARY_DIR MATCHES "${CMAKE_SOURCE_DIR}")
+    # if CMAKE_BINARY_DIR is inside CMAKE_SOURCE_DIR Ninja uses relative paths which screws logging of the filename
+    # so we just trim the first "../" here, this is not always correct but will show proper file
+    add_definitions("-DSOURCE_PATH_SIZE=3")
+else()
+    message("binary dir = ${CMAKE_BINARY_DIR}")
```

Comment:
did you leave this message in here intentionally? why only for the 'else'?

Reply:
it's just debugging, I'll remove